index.html

```diff
@@ -2107,6 +2121,105 @@ <h3>Securing Verifiable Credentials</h3>
 Directory [[VC-SPECS]].
         </p>
 
+        <section class="normative">
+          <h4>The Verification Algorithm</h4>
+
+          <p>
+This section contains an algorithm that <a>conforming verifier processors</a>
+can run when verifying a <a>verifiable credential</a> or a <a>verifiable
+presentation</a>. This algorithm takes a byte sequence (bytes
+<var>inputBytes</var>) and a media type (string <var>inputMediaType</var>) as
+inputs, and produces an object (<var>verified</var>) that contains a
+verification status (boolean <var>verified</var>.<var>status</var>), a verified
+document (object <var>verified</var>.<var>document</var>), a media type (string
+<var>verified</var>.<var>mediaType</var>), zero or more optional warnings (array
+of <a>anomalyDescriptions</a> <var>verified</var>.<var>warnings</var>), and zero
+or more optional errors (array of <a>anomalyDescriptions</a>
+<var>verified</var>.<var>errors</var>).
+          </p>
+
+          <ol class="algorithm">
+            <li>
+Set <var>verified</var> to an empty object.
+            </li>
+            <li>
+Verify that the securing mechanism in use on <var>inputBytes</var> has not been
+tampered with, using <var>inputMediaType</var> as additional input to the
+algorithm. See Section <a href="#securing-mechanisms"></a>
+for details.
+            </li>
+            <li>
+If the verification of the securing mechanism is successful, set
+<var>verified</var>.<var>status</var> to `true`,
+<var>verified</var>.<var>document</var> to the information that was checked and
+returned as successfully protected by the verification process for the securing
+mechanism as, <var>verified</var>.<var>mediaType</var> to the corresponding
+media type, and optionally <var>verified</var>.<var>warnings</var> to any
+warnings returned by the verification process for the securing mechanism.
+              <ol class="algorithm">
+                <li>
+Check whether <var>verified</var>.<var>document</var> is a
+<a>conforming document</a>. If not, optionally add appropriate warnings
```

Comment:
Oh, based on the processing model described in https://github.com/w3c/vc-data-model/pull/1349#discussion_r1391685319, some of these conformance checks need to happen before the securing mechanism gets called.

For an RDF-based securing mechanism, what's described here has the securing mechanism parse the document and then re-serialize part of it, and that serialized version is what has "use `id` and no other shorthand for `@id`" enforced. But that tests the securing mechanism's serialization code, not the input VC.

No need to fix this before merging this initial algorithm.

Reply:
> Oh, based on the processing model described in [#1349 (comment)](https://github.com/w3c/vc-data-model/pull/1349#discussion_r1391685319), some of these conformance checks need to happen before the securing mechanism gets called.

Yes, there is debate on when the appropriate time to do this is (before or after you've checked the securing mechanism). AFAIK, most implementations do the conforming document checks before checking the signature (via the use of JSON Schema or hard-coded checks). It's conceivable that doing the check after (based only on the secured information) is another valid way to do this as well. There are good arguments for/against each approach; I'm on the fence on which one is the better approach, so strong opinions on the right thing to do would be welcome here.

> For an RDF-based securing mechanism, what's described here has the securing mechanism parse the document and then re-serialize part of it, and that serialized version is what has "use `id` and no other shorthand for `@id`" enforced. But that tests the securing mechanism's serialization code, not the input VC.

Yes, but that's not the case of JOSE-based mechanisms. The thing that matters here is that the application is working w/ only the secured information after the securing mechanism has been checked, and the information it's working with is a conforming document. The order in which those operations happen is, arguably, not as important as the end result.

> No need to fix this before merging this initial algorithm.

Thanks for the flexibility. We will, however, need to have a very good basis for why we do one before the other, IMHO... or, come to the conclusion that the order that the checks are done is up to the implementation, as long as the end result is that you're working with the information that was successfully secured and verified, and that information is a conforming document.